game-core/src/test/java/games/strategy/engine/message/unifiedmessenger/EndPointTest.java

```diff
@@ -2,25 +2,26 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
+import games.strategy.engine.message.RemoteActionCode;
 import games.strategy.engine.message.RemoteMethodCall;
 import games.strategy.engine.message.RemoteMethodCallResults;
-import java.util.Comparator;
 import java.util.List;
 import org.junit.jupiter.api.Test;
 
 class EndPointTest {
 
+  interface TestInterface {
+    @RemoteActionCode(0)
+    @SuppressWarnings("unused")
+    int dummy();
+  }
+
   @Test
-  void testEndPoint() {
-    final EndPoint endPoint = new EndPoint("", Comparator.class, false);
-    endPoint.addImplementor((Comparator<Object>) (o1, o2) -> 2);
+  void testEndPoint() throws Exception {
+    final EndPoint endPoint = new EndPoint("", TestInterface.class, false);
+    endPoint.addImplementor((TestInterface) () -> 2);
     final RemoteMethodCall call =
-        new RemoteMethodCall(
-            "",
-            "compare",
-            new Object[] {"", ""},
-            new Class<?>[] {Object.class, Object.class},
-            Comparator.class);
+        new RemoteMethodCall("", TestInterface.class.getMethod("dummy"), new Object[] {});
```

Comment:
Looks like we are still bound by method name (and ordering must be preserved, but at least that is in annotation now). Is there a way we could put the method name in the annotation as well?

Reply:
We are only bound in this particular test-case.
The RemoteMethodCall class works via reflection after all, so we need to pass the method object to the constructor somehow.
In non-test code this conversion is done by the JVM:
We create a "proxy" object that implements a particular interface. Whenever we call a method on this proxy object, the invocationhandler we registered is called with the corresponding method object, so the non-test code doesn't have any references to the method names whatsoever